zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -2271,7 +2271,14 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
   });
 
   $scope.$on('appendParagraphOutput', function(event, data) {
-    if ($scope.paragraph.id === data.paragraphId) {
+    /* It has been observed that append events
+     * can be errorneously called even if paragraph
+     * execution has ended, and in that case, no append
+     * should be made. Also, it has been observed that this
+     * could be called in PENDING state as well
+     */
+    if ($scope.paragraph.id === data.paragraphId &&
+       ($scope.paragraph.status === 'RUNNING' || $scope.paragraph.status === 'PENDING')) {
```

Comment:
I see, the comment was a bit misleading, as it seems it was saying that it could be errorneously call es as well in PENDING state.